File: 2021_Kakao_Internship/Hoovely/숫자 문자열과 영단어.py
def solution(s):
    result = ""
    num = ""
    nums = {'zero': '0', 'one': '1', 'two': '2', 'three': '3', 'four': '4',
            'five': '5', 'six': '6', 'seven': '7', 'eight': '8', 'nine': '9'}
    for i in s:
        if i.isdecimal():
            result += i
        else:
            num += i
            if num in nums:
                result += nums[num]
                num = ""
    return int(result)

File: 2021_Kakao_Internship/Alcholrithm/숫자_문자열과_영단어.cpp
#include <vector>
#include <string>
#include <map>

using namespace std;

int solution(string s) {
    int answer = 0;
    map<string,  int> m;
    m["zero"] = 0;
    m["one"] = 1;
    m["two"] = 2;
    m["three"] = 3;
    m["four"] = 4;
    m["five"] = 5;
    m["six"] = 6;
    m["seven"] = 7;
    m["eight"] = 8;
    m["nine"] = 9;

    string temp;
    for(int i = 0; i < s.size(); ++i) {
        if (s[i] >= '0' && s[i] <= '9') {
            answer = answer * 10 + s[i] - '0';
        }
        else {
            temp += s[i];
            if (m.find(temp) != m.end()) {
                answer = answer * 10 + m[temp];
                temp.clear();
            }
        }
    }
    return answer;
}